confluent_kafka/serialization/kafka_serdes.py

```diff
@@ -0,0 +1,343 @@
+
+    """  # noqa: E501
+    def __call__(self, datum, ctx):
+        """
+        Serializes int as int32 bytes.
+
+        Args:
+            datum (int): int to be serialized.
+            ctx (SerializationContext): Serialization context
+
+        Raises:
+            SerializerError if an error occurs daring serialization
+
+        Returns:
+            int32 bytes if datum is not None, else None
+
+        """
+        if datum is None:
+            return None
+
+        try:
+            return struct.pack('>i', datum)
+        except struct.error as e:
+            raise SerializationError(str(e))
+
+
+class IntegerDeserializer(Deserializer):
+    """
+    Deserializes int to int32 bytes.
+
+    .._IntegerDeserializer:
+        https://docs.confluent.io/current/clients/javadocs/org/apache/kafka/common/serialization/IntegerDeserializer.html
+
+    """  # noqa: E501
+    def __call__(self, data, ctx):
+        """
+        Deserializes int from int32 bytes.
+
+        Args:
+            data(bytes): int32 bytes
+            ctx (SerializationContext): Serialization context
+
+        Raises:
+            SerializerError if an error occurs daring deserialization.
+
+        Returns:
+            int if data is not None, otherwise None
+
+        """
+        if data is None:
+            return None
+
+        try:
+            return struct.unpack('>i', data)[0]
+        except struct.error as e:
+            raise SerializationError(str(e))
+
+
+class StringSerializer(Serializer):
+    """
+    Serializes unicode to bytes per the configured codec. Defaults to ``utf_8``.
+
+    Keyword Args:
+        - codec (str, optional): encoding scheme. Defaults to utf_8
+
+    .. _StandardEncodings:
+        https://docs.python.org/3/library/codecs.html#standard-encodings
+
+    .. _StringSerializer:
+        https://docs.confluent.io/current/clients/javadocs/org/apache/kafka/common/serialization/StringSerializer.html
+
+    """  # noqa: E501
+    def __init__(self, codec='utf_8'):
+        self.codec = codec
+
+    def __call__(self, datum, ctx):
+        """
+        Serializes a str(py2:unicode) to bytes.
+
+        Compatibility Note:
+            Python 2 str objects must be converted to unicode objects.
+            Python 3 all str objects are already unicode objects.
+
+        Args:
+            datum (unicode): Unicode object to serialize
+            ctx (SerializationContext): Serialization context
+
+        Raises:
+            SerializerError if an error occurs daring serialization.
+
+        Returns:
+            serialized bytes if datum is not None, otherwise None
+
+        """
+        if datum is None:
+            return None
+
+        try:
+            return datum.encode(self.codec)
+        except struct.error as e:
+            raise SerializationError(str(e))
+
+
+class StringDeserializer(Deserializer):
+    """
+    Deserializes utf-8 bytes to str(py2:unicode) object.
```

Comment:
utf-8 is configurable, so should probably say something more generic here.
Compare ot StringSerializer